int32  decv_id
int32  decv_state